tests/unit/unit-util.cpp

```diff
@@ -229,6 +229,38 @@ TEST_P(UnitUtilDimacsTests, cnf_to_dimacs)
 }
 
 
+
+TEST_P(UnitUtilDimacsTests, tseitin)
+{
+  s->set_opt("incremental", "true");
+  Term p = s->make_symbol("p", boolsort);
+  Term q = s->make_symbol("q", boolsort);
+  Term r = s->make_symbol("r", boolsort);
+  Term t = s->make_symbol("t", boolsort);
+
+  //a=((p or q) and r) implies (not t)
+  Term a = s->make_term(Implies, s->make_term(And, s->make_term(Or, p, q), r), s->make_term(Not, t));
+  Term cnf1 = to_cnf(a, s);
+  s->assert_formula(cnf1);
+  Result r1 = s->check_sat();
+  ASSERT_TRUE(r1.is_sat());
+  //b=(not (p xor q))
+  Term b = s->make_term(Not, s->make_term(Xor, p, q));
+  Term cnf2 = to_cnf(b, s);
+  s->assert_formula(cnf2);
+  Result r2 = s->check_sat();
+  ASSERT_TRUE(r2.is_sat());
+  //c=((not p) and p)
+  Term c = s->make_term(And, s->make_term(Not, p), p);
+  Term cnf3 = to_cnf(c, s);
+  s->assert_formula(cnf3);
+  Result r3 = s->check_sat();
+  ASSERT_FALSE(r3.is_sat());
+
```

Comment:
The actual cnf had become too long, so I thought it would be unreadable. I had checked that it was exactly what I had expected. I'll add it as a test though.